programs/stake/src/config.rs

```diff
@@ -1,51 +1,32 @@
 //! config for staking
 //!  carries variables that the stake program cares about
-use bincode::{deserialize, serialized_size};
-use serde_derive::{Deserialize, Serialize};
-use solana_config_program::{create_config_account, get_config_data, ConfigState};
+use bincode::deserialize;
+use solana_config_program::{create_config_account, get_config_data};
 use solana_sdk::{
     account::{AccountSharedData, ReadableAccount, WritableAccount},
     genesis_config::GenesisConfig,
     instruction::InstructionError,
     keyed_account::KeyedAccount,
+    stake::config::{self, Config},
 };
 
-// stake config ID
-solana_sdk::declare_id!("StakeConfig11111111111111111111111111111111");
-
-// means that no more than RATE of current effective stake may be added or subtracted per
-//  epoch
-pub const DEFAULT_WARMUP_COOLDOWN_RATE: f64 = 0.25;
-pub const DEFAULT_SLASH_PENALTY: u8 = ((5 * std::u8::MAX as usize) / 100) as u8;
-
-#[derive(Serialize, Deserialize, Debug, PartialEq, Clone, Copy)]
-pub struct Config {
-    /// how much stake we can activate/deactivate per-epoch as a fraction of currently effective stake
-    pub warmup_cooldown_rate: f64,
-    /// percentage of stake lost when slash, expressed as a portion of std::u8::MAX
-    pub slash_penalty: u8,
-}
-
-impl Config {
-    pub fn from<T: ReadableAccount>(account: &T) -> Option<Self> {
+pub struct ConfigConverter;
```

Comment:
What's the motivation behind this, and all the associated functions? I might be missing something downstream, but it seems like `from_keyed_account()` and `create_account()` didn't need to be changed, and `Config::from()` could just become `config::from()`

Reply:
Since bpf programs don't have access to `KeyedAccount` from the sdk, I decided to keep all code that uses them in the `programs/stake` code, and combine all of the converters into a namespace whose name won't conflict with the types now in `solana_program`.  The intention wasn't to change any of it, so if anything has changed logic-wise, please let me know!  Moving `config::from()` and friends into a struct was entirely an aesthetics move to make the imports look prettier in test code.  If you think it's better to keep all of those functions at the top-level of the file, I can move them back out.

Reply:
I'm just now remembering more specifically why I did this.  Since a lot of the functions were part of the `Stake` or `Config` structs, I wanted to keep them as part of a struct to make the change look as small as possible and to make the diff easier to read.

Reply:
Thanks for the context. I guess my personal aesthetic preference would be to keep the functions at the top level of the module.
Also, fwiw it looks to me like there's actually more diff with this change, eg this, which wouldn't otherwise need to be touched:
```diff
-               &config::from_keyed_account(keyed_account_at_index(keyed_accounts, 4)?)?,
+               &ConfigConverter::from_keyed_account(keyed_account_at_index(keyed_accounts, 4)?)?,
```

Reply:
Fair enough, I'll move those functions out :smile: